django/contrib/auth/middleware.py

```diff
@@ -34,6 +39,68 @@ def process_request(self, request):
         request.auser = partial(auser, request)
 
 
+class LoginRequiredMiddleware:
+    """
+    Middleware that forces all views to require login by default.
+
+    Views that have login_not_required decorator or LoginNotRequiredMixin mixin
+    will be able to pass through without this validation. Otherwise, it will
+    redirect user to login or admin's login if accessing admin site.
+    """
+
+    sync_capable = True
+    async_capable = True
+
+    def __init__(self, get_response):
+        if get_response is None:
+            raise ValueError("get_response must be provided.")
+        self.get_response = get_response
+        self.async_mode = iscoroutinefunction(self.get_response)
+        if self.async_mode:
+            markcoroutinefunction(self)
+        super().__init__()
+
+    def __call__(self, request):
+        if not hasattr(request, "user"):
+            raise ImproperlyConfigured(
+                "The Django login required middleware requires authentication "
+                "middleware to be installed. Edit your MIDDLEWARE setting to "
+                "insert "
+                "'django.contrib.auth.middleware.AuthenticationMiddleware' "
+                "before "
+                "'django.contrib.auth.middleware.LoginRequiredMiddleware'."
+            )
```

Comment:
I don't think we need this error. We don’t do equivalent checks in `LoginRequiredMixin`, for example.

To defend against the case where the auth middleware is missing or out of order, it would be better done in a system check. That way there won’t be any extra work on the per-request path. The admin system checks have a small example of checking the middleware setting.

Reply:
Added it to system check.